src/vsr/grid_blocks_missing.zig

```diff
@@ -329,19 +389,30 @@ pub const GridBlocksMissing = struct {
         }
     }
 
+    pub fn repairing_blocks(queue: *const GridBlocksMissing) bool {
+        return queue.repairing_tables() or queue.enqueued_blocks_single > 0;
+    }
+
+    pub fn repairing_tables(queue: *const GridBlocksMissing) bool {
+        return queue.state != .syncing and queue.enqueued_blocks_table > 0;
+    }
+
     pub fn repair_waiting(queue: *const GridBlocksMissing, address: u64, checksum: u128) bool {
         const fault_index = queue.faulty_blocks.getIndex(address) orelse return false;
         const fault = &queue.faulty_blocks.values()[fault_index];
         return fault.checksum == checksum and fault.state == .waiting;
     }
```

Comment:
One alternate approach I tried to avoid double-buffering the `faulty_blocks` hashmap is: have `repair_waiting` return false for table blocks when `state=syncing`.

But that doesn't work -- while syncing, we might try to queue a single block (i.e. a trailer) with the same address as a soon-to-be-canceled table block, causing a collision. We _could_ detect that collision and evict the table at that point. But that handle all the cancellation at once, to avoid a hard-to-test special case + added dimensionality.